h5py/_hl/dataset.py

```diff
@@ -100,10 +100,12 @@ def make_new_dset(parent, shape=None, dtype=None, data=None, name=None,
             raise TypeError("Conflict in compression options")
         compression_opts = compression
         compression = 'gzip'
-    dcpl = filters.fill_dcpl(
-        dcpl or h5p.create(h5p.DATASET_CREATE), shape, dtype,
```

Comment:
I actually quite like the `x or make_default()` pattern for things like this.

Reply:
Mm-hmm, then I will revert this part.